src/nats/connection.cr

```diff
@@ -511,7 +515,9 @@ module NATS
       cs[:user] = @user.to_s if @user
       cs[:pass] = @pass.to_s if @pass
 
-      @socket << "CONNECT #{cs.to_json}\r\n"
+      @socket << "CONNECT "
+      cs.to_json(@socket)
+      @socket << "\r\n"
```

Comment:
Actually in this method we can completely avoid creating a Hash, and write the JSON directly to the IO, but since this is just in connect it might not be worth it :-)

(in fact changing the interpolation to what I did might not be needed, I just now realized this is not in a tight loop)

Reply:
Well, I did it anyway because it's more or less the same amount of code and it should be slightly faster.

Reply:
Very cool!